### ISSUE TITLE
requests.exceptions.MissingSchema: Invalid URL '334fmf.json': No schema supplied.

### ISSUE THREAD

**Issue by tek0011** [tek0011/reddit_info_bot#59]
```
finding username mentions: Image-searching for http://i.imgur.com/mQ7Tuye.gifv
Found imgur.com video - substituting with gif url: http://i.imgur.com/mQ7Tuye.gif
Traceback (most recent call last):
  File "info_bot.py", line 4, in <module>
    import reddit_info_bot
  File "/usr/reddit_info_bot/reddit_info_bot/__init__.py", line 498, in <module>
    main()
  File "/usr/reddit_info_bot/reddit_info_bot/__init__.py", line 423, in main
    find_username_mentions()
  File "/usr/reddit_info_bot/reddit_info_bot/__init__.py", line 271, in find_username_mentions
    reply = give_more_info(comment.submission.url, display_limit=5)
  File "/usr/reddit_info_bot/reddit_info_bot/__init__.py", line 188, in give_more_info
    filtered  = _filter_results(result, account1, account2, config['SUBMISSION_ID'])
  File "/usr/reddit_info_bot/reddit_info_bot/__init__.py", line 67, in _filter_results
    results = _reddit_spamfilter(results, account2, account1, check_submission_id)
  File "/usr/reddit_info_bot/reddit_info_bot/__init__.py", line 41, in _reddit_spamfilter
    verified_urls = reddit_msg_linkfilter(urls, sending_account, receiving_account, submission_id)
  File "/usr/reddit_info_bot/reddit_info_bot/reddit.py", line 37, in reddit_msg_linkfilter
    submission = sending_account.get_submission(submission_id)
  File "/usr/local/lib/python2.7/site-packages/praw/__init__.py", line 1043, in get_submission
    params=params)
  File "<decorator-gen-191>", line 2, in from_url
  File "/usr/local/lib/python2.7/site-packages/praw/decorators.py", line 268, in wrap
    return function(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/praw/objects.py", line 1111, in from_url
    response = reddit_session.request_json(url, params=params)
  File "<decorator-gen-8>", line 2, in request_json
  File "/usr/local/lib/python2.7/site-packages/praw/decorators.py", line 113, in raise_api_exceptions
    return_value = function(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/praw/__init__.py", line 612, in request_json
    retry_on_error=retry_on_error)
  File "/usr/local/lib/python2.7/site-packages/praw/__init__.py", line 444, in _request
    response = handle_redirect()
  File "/usr/local/lib/python2.7/site-packages/praw/__init__.py", line 422, in handle_redirect
    request=request.prepare(),
  File "/usr/local/lib/python2.7/site-packages/requests/models.py", line 249, in prepare
    hooks=self.hooks,
  File "/usr/local/lib/python2.7/site-packages/requests/models.py", line 293, in prepare
    self.prepare_url(url, params)
  File "/usr/local/lib/python2.7/site-packages/requests/models.py", line 353, in prepare_url
    raise MissingSchema(error)
requests.exceptions.MissingSchema: Invalid URL '334fmf.json': No schema supplied. Perhaps you meant http://334fmf.json?
```

**Comment by nyov** [comment]
wtf, praw...?
Is this a recurring problem? I'm not quite sure how this happens, either the `submission_id` must be missing at that point, or praw just somehow failed to build a valid URL.

**Comment by tek0011** [comment]
not sure.  Let me try another test.

On Thu, Oct 8, 2015 at 1:38 PM, nyov <notifications@github.com> wrote:

> wtf, praw...?
> Is this a recurring problem? I'm not quite sure how this happens, either
> the submission_id must be missing at that point, or praw just somehow
> failed to build a valid URL.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tek0011/reddit_info_bot/issues/59#issuecomment-146648667>
> .
>



-- 
Jeff

**Comment by tek0011** [comment]
exact same error on every request

On Thu, Oct 8, 2015 at 1:50 PM, Jeff <thecptsparrow@gmail.com> wrote:

> not sure.  Let me try another test.
>
> On Thu, Oct 8, 2015 at 1:38 PM, nyov <notifications@github.com> wrote:
>
>> wtf, praw...?
>> Is this a recurring problem? I'm not quite sure how this happens, either
>> the submission_id must be missing at that point, or praw just somehow
>> failed to build a valid URL.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/tek0011/reddit_info_bot/issues/59#issuecomment-146648667>
>> .
>>
>
>
>
> --
> Jeff
>



-- 
Jeff

**Comment by nyov** [comment]
I'll investigate. In case it might matter, whats the PRAW version?
In the meantime that spam-check could be skipped/deactivated by removing the second account user or pw from the config.

**Comment by tek0011** [comment]
I completely uninstall praw, and reinstalled.  same issue

On Thu, Oct 8, 2015 at 2:12 PM, nyov <notifications@github.com> wrote:

> I'll investigate. In case it might matter, whats the PRAW version?
> In the meantime that spam-check could be skipped/deactivated by removing
> the second account user or pw from the config.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tek0011/reddit_info_bot/issues/59#issuecomment-146659247>
> .
>



-- 
Jeff

**Comment by tek0011** [comment]
oh sorry.  missed your email.  its 3.3.0

On Thu, Oct 8, 2015 at 2:18 PM, Jeff <thecptsparrow@gmail.com> wrote:

> I completely uninstall praw, and reinstalled.  same issue
>
> On Thu, Oct 8, 2015 at 2:12 PM, nyov <notifications@github.com> wrote:
>
>> I'll investigate. In case it might matter, whats the PRAW version?
>> In the meantime that spam-check could be skipped/deactivated by removing
>> the second account user or pw from the config.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/tek0011/reddit_info_bot/issues/59#issuecomment-146659247>
>> .
>>
>
>
>
> --
> Jeff
>



-- 
Jeff

**Comment by nyov** [comment]
Ah, found it. Must have removed a positional keyword, it was looking for a full URL instead of the `submission_id`.

**Comment by nyov** [comment]
Should be fixed by #60
